src/ripple/rpc/handlers/AccountObjects.cpp

```diff
@@ -72,12 +72,49 @@ Json::Value doAccountObjects (RPC::Context& context)
     if (! ledger->exists(keylet::account (accountID)))
         return rpcError (rpcACT_NOT_FOUND);
 
-    auto [rpcStatus, type] = RPC::chooseLedgerEntryType(params);
-    if (rpcStatus)
+    std::optional<std::vector<LedgerEntryType>> typeFilter;
+
+    if (params.isMember(jss::deletion_blockers_only) &&
+        params[jss::deletion_blockers_only].asBool())
     {
-        result.clear();
-        rpcStatus.inject(result);
-        return result;
+        struct NameType {
+            Json::StaticString name;
+            LedgerEntryType type;
+        } static constexpr deletionBlockers[] = {
+            { jss::check,           ltCHECK },
+            { jss::escrow,          ltESCROW },
+            { jss::payment_channel, ltPAYCHAN },
+            { jss::state,           ltRIPPLE_STATE }
+        };
+
+        typeFilter.emplace();
+        typeFilter->reserve(std::size(deletionBlockers));
+
+        for (auto [name, type] : deletionBlockers)
+        {
+            if (params.isMember(jss::type) &&
+                name != params[jss::type])
+            {
+                continue;
+            }
+
+            typeFilter->push_back(type);
+        }
+    }
+    else
+    {
+        auto [rpcStatus, type] = RPC::chooseLedgerEntryType(params);
+
+        if (rpcStatus)
+        {
+            result.clear();
+            rpcStatus.inject(result);
+            return result;
+        }
+        else if (type != ltINVALID)
+        {
+            typeFilter.emplace({ type });
```

Comment:
If we change `typeFilter` to `boost::optional` we need to change this to  `typeFilter.emplace(type);`

Reply:
That ended up calling the wrong constructor for `std::vector`, the `type` enum implicitly converted to a `size_t`. Should be good now.